File: src/main/java/net/hectus/neobb/turn/default_game/block/TLava.java
package net.hectus.neobb.turn.default_game.block;

import net.hectus.neobb.buff.Buff;
import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.attributes.clazz.HotClazz;
import net.hectus.neobb.turn.default_game.attributes.function.BuffFunction;
import org.bukkit.Material;
import org.bukkit.block.Block;
import org.bukkit.inventory.ItemStack;

import java.util.List;

public class TLava extends BlockTurn implements BuffFunction, HotClazz {
    public TLava(NeoPlayer player) { super(player); }
    public TLava(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public boolean requiresUsageGuide() { return true; }

    @Override
    public ItemStack item() {
        return new ItemStack(Material.LAVA_BUCKET);
    }

    @Override
    public int cost() {
        return 6;
    }

    @Override
    public void apply() {
        player.nextPlayer().player.setFireTicks(6000); // 5 Minutes Max
        player.game.turnScheduler.runTaskLater("burn", () -> player.game.eliminatePlayer(player.nextPlayer()), 3);
    }

    @Override
    public int maxAmount() { return 1; }

    @Override
    public List<Buff> buffs() { return List.of(); }
}

File: src/main/kotlin/net/hectus/neobb/turn/default_game/block/TLava.kt
package net.hectus.neobb.turn.default_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.buff.Buff
import net.hectus.neobb.game.util.ScheduleID
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.turn.default_game.attribute.clazz.HotClazz
import net.hectus.neobb.turn.default_game.attribute.function.BuffFunction
import org.bukkit.Material
import org.bukkit.block.Block
import org.bukkit.inventory.ItemStack

class TLava(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), BuffFunction, HotClazz {
    override val cost: Int = 6
    override val requiresUsageGuide: Boolean = true
    override val maxAmount: Int = 1

    override fun item(): ItemStack = ItemStack(Material.LAVA_BUCKET)

    override fun apply() {
        player!!.nextPlayer().player.fireTicks = 6000
        player.game.turnScheduler.runTaskLater(ScheduleID.BURN, 3) {
            player.game.eliminate(player.nextPlayer())
        }
    }

    override fun buffs(): List<Buff<*>> = emptyList()
}